scipy/stats/_continuous_distns.py

```diff
@@ -8806,6 +8806,135 @@ def _entropy(self, kappa):
         return (-kappa * sc.i1(kappa) / sc.i0(kappa) +
                 np.log(2 * np.pi * sc.i0(kappa)))
 
+    def expect(self, func=None, args=(), loc=0, scale=1, lb=-np.pi, ub=np.pi,
+                conditional=False, **kwds):
+        """Calculate expected value of a function with respect to the
+        distribution by numerical integration.
+
+        The expected value of a function ``f(x)`` with respect to a
+        distribution ``dist`` is defined as::
+
+                    ub
+            E[f(x)] = Integral(f(x) * dist.pdf(x)),
+                    lb
+
+        where ``ub`` and ``lb`` are arguments and ``x`` has the ``dist.pdf(x)``
+        distribution. If the bounds ``lb`` and ``ub`` correspond to the
+        support of the distribution, e.g. ``[-pi, pi]`` (or any interval
+        spanning ``2pi``) in the default
+        case, then the integral is the unrestricted expectation of ``f(x)``.
+        Also, the function ``f(x)`` may be defined such that ``f(x)`` is ``0``
+        outside a finite interval in which case the expectation is
+        calculated within the finite range ``[lb, ub]``.
+
+        Parameters
+        ----------
+        func : callable, optional
+            Function for which integral is calculated. Takes only one argument.
+            The default is the identity mapping f(x) = x.
```

Comment:
default here is not identity mapping anymore

Reply:
also it should mention that complex return of func is supported

Reply:
This should be resolved in the latest commits to this PR.